akka-http2-support/src/test/java/com/twitter/hpack/HexUtil.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twitter.hpack;
+
+import java.io.IOException;
+
+/**
+ * Extracted from org/apache/commons/codec/binary/Hex.java
+ * Copyright Apache Software Foundation
+ */
+final public class HexUtil {
+
+  /**
+   * Used to build output as Hex
+   */
+  private static final char[] DIGITS_LOWER = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+
+  /**
+   * Used to build output as Hex
+   */
+  private static final char[] DIGITS_UPPER = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
+
+  /**
+   * Converts an array of characters representing hexadecimal values into an array of bytes of those same values. The
+   * returned array will be half the length of the passed array, as it takes two characters to represent any given
+   * byte. An exception is thrown if the passed char array has an odd number of elements.
+   *
+   * @param data
+   *            An array of characters containing hexadecimal digits
+   * @return A byte array containing binary data decoded from the supplied char array.
+   * @throws IOException
+   *             Thrown if an odd number or illegal of characters is supplied
+   */
+  public static byte[] decodeHex(char[] data) throws IOException {
```

Comment:
![Calmer than you are](https://s-media-cache-ak0.pinimg.com/564x/7a/48/c9/7a48c9bbc6e817ed8e3dd44c3efd5c00.jpg)